pkg/virt-handler/cmd-client/client.go

```diff
@@ -421,6 +431,48 @@ func (c *VirtLauncherClient) MigrateVirtualMachine(vmi *v1.VirtualMachineInstanc
 
 }
 
+func (c *VirtLauncherClient) HandleDomainEvents() error {
+	cmd, err := c.v1client.HandleDomainEvent(context.Background(), &cmdv1.EmptyRequest{})
+	if err != nil {
+		return err
+	}
+	for {
+		msg, err := cmd.Recv()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		err = common.EnqueueHandlerDomainEvent(c.eventChan, msg)
```

Comment:
where would you filter to ensure only events pertaining to this client can be sent?

Reply:
I think that we can keep enough metadata on each client, so that we can pass in a vmi/domain reference to this function. In that function we can then filter unwanted events.